### ISSUE TITLE
Fails on labels setup if hostname=fqdn

### ISSUE THREAD

**Issue by evgeniy-pupkov** [atosatto/ansible-dockerswarm#61]
Role fails on "Get list of labels" task if target hosts hostname is FQDN with 'no such object ' error

**Comment by atosatto** [comment]
Hi @evgeniy-pupkov, can you please share an example of your role configuration?
We're working on a 2.0 release of the role and we would like to make sure that this issue is addressed.

**Comment by mightydok** [comment]
If docker info show Name: FQDN, this role will fail.
Because in setup-swarm-labels you use ansible_hostname, not ansible_fqdn

**Comment by atosatto** [comment]
Ehi @mightydok are you suggesting we should replace `ansible_hostname` with `ansible_fqdn` in `tasks/setup-swarm-labels.yaml`?

**Comment by wombathuffer** [comment]
I'm stuck on same task. Havent figured yet how to work around it though..
`TASK [atosatto.docker-swarm : Get list of labels.] ***************************************************************************
task path: /repos/yi-ansible/roles/atosatto.docker-swarm/tasks/setup-swarm-labels.yml:3
fatal: [asus.yi -> None]: FAILED! => {"changed": false, "cmd": ["docker", "inspect", "--format", "{{ range $key, $value := .Spec.Labels }}{{ printf \"%s\\n\" $key }}{{ end }}", "asus"], "delta": "0:00:00.062732", "end": "2020-03-15 06:47:44.650009", "msg": "non-zero return code", "rc": 1, "start": "2020-03-15 06:47:44.587277", "stderr": "Error: No such object: asus", "stderr_lines": ["Error: No such object: asus"], "stdout": "", "stdout_lines": []}
`


This is my inventory:
```yaml
all:
  children:
    docker_swarm:
      hosts:
        asus.yi:
          ansible_ssh_host: 10.0.0.7
    docker_engine:
      hosts:
        asus:
    docker_swarm_manager:
      hosts:
        asus.yi:
          swarm_labels: deploy
    docker_swarm_worker:
      hosts:
        asus.yi:
          swarm_labels: ["centos","foo","bar"]
```
The result is labels are not populated.
```bash
[root@asus ~]# docker inspect --format '{{json .Spec.Labels}}' asus.yi | jq
{}
[root@asus ~]# hostname -f
asus.yi
```

How do I bypass this? :)

**Comment by wombathuffer** [comment]
The solution for me, was to replace `{{ ansible_hostname }}` with `{{ ansible_fqdn|lower }}` in `tasks/setup-swarm-labels.yaml`:
```yaml
---

- name: Get list of labels.
  command: >-
         docker inspect
         --format {% raw %}'{{ range $key, $value := .Spec.Labels }}{{ printf "%s\n" $key }}{{ end }}'{% endraw %}
         {{ ansible_fqdn|lower }}
  register: docker_swarm_labels
  changed_when: false
  delegate_to: "{{ groups['docker_swarm_manager'][0] }}"
  delegate_facts: true
  tags:
    - swarm_labels

- name: Remove labels from swarm node.
  command: docker node update --label-rm {{ item }} {{ ansible_fqdn|lower }}
  with_items: "{{ docker_swarm_labels.stdout_lines }}"
  when: item not in swarm_labels
  delegate_to: "{{ groups['docker_swarm_manager'][0] }}"
  delegate_facts: true
  tags:
    - swarm_labels

- name: Assign labels to swarm nodes if any.
  command: docker node update --label-add {{ item }}=true {{ ansible_fqdn|lower }}
  when: item not in docker_swarm_labels.stdout_lines
  with_items:
    - "{{ swarm_labels  | default([]) }}"
  delegate_to: "{{ groups['docker_swarm_manager'][0] }}"
  delegate_facts: true
  tags:
    - swarm_labels

```

**Comment by Koma-Andrea** [comment]
I can confirm that {{ ansible_fqdn|lower }} is the right solution.
I'll build a PR

**Comment by atosatto** [comment]
Thanks Andrea!
I will review thar immediately so that we can finally close this
longstanding issue.

**Comment by Koma-Andrea** [comment]
Pull request #75 .
BTW Good  Role! I appreciate it!

**Comment by atosatto** [comment]
Fixed in #75. Thanks @Koma-Andrea! :)

**Comment by atosatto** [comment]
I've just released [v2.3.0](https://github.com/atosatto/ansible-dockerswarm/releases/tag/v2.3.0) which is including a fix to this issue. Thanks everyone for investigating and helping fixing it :)
